.github/workflows/check-motoko-changes.yml

```diff
@@ -32,25 +36,33 @@ jobs:
           HEAD_SHA="${{ github.event.pull_request.head.sha }}"
 
           # Check for any .mo changes
-          if ! git diff --name-only $BASE_SHA $HEAD_SHA | grep -q '\.mo$'; then
+          if [ -z "$(git diff --name-only $BASE_SHA $HEAD_SHA -- '*.mo')" ]; then
             echo "has_mo_changes=false" >> $GITHUB_OUTPUT
             echo "No Motoko file changes detected"
           else
             echo "has_mo_changes=true" >> $GITHUB_OUTPUT
           fi
 
           # Check for src/*.mo changes
-          if ! git diff --name-only $BASE_SHA $HEAD_SHA | grep -q '^src/.*\.mo$'; then
+          if [ -z "$(git diff --name-only $BASE_SHA $HEAD_SHA -- 'src/*.mo')" ]; then
```

Comment:
Does really work? TIL, you can specify patterns there (or is that just shell expansion?). But then do we still need `--name-only`?

Reply:
`--name-only` is not necessary, just makes for a shorter output.
`git diff` always returns 0, so I needed to compare its output whether it's empty.
Yes, patterns like `'src/*.mo'` work, this finds all `.mo` files under the `src/` directory in the root project